lightning/src/chain/keysinterface.rs

```diff
@@ -1074,12 +1080,17 @@ impl KeysManager {
 					}
 					spend_tx.input[input_idx].witness = keys_cache.as_ref().unwrap().0.sign_dynamic_p2wsh_input(&spend_tx, input_idx, &descriptor, &secp_ctx)?;
 				},
-				SpendableOutputDescriptor::StaticOutput { ref output, .. } => {
+				SpendableOutputDescriptor::StaticOutput { ref output, channel_keys_id,.. } => {
+					
+					let destination_script = channel_keys_id.map_or_else(|| self.destination_script.clone(),|s| self.get_destination_script(s));
```

Comment:
It looks like this is unused for the `!channel_keys_id.is_some()`, maybe just `map` and leave it as `None`, then in the if below do `if Some(&output.script_pubkey) == destination_script.as_ref()`